### ISSUE TITLE
Add Limit to object-property-newline allowAllPropertiesOnSameLine 

### ISSUE THREAD

**Issue by Rich-PF** [eslint/eslint#13565]
**What rule do you want to change?**
object-property-newline, allowAllPropertiesOnSameLine 

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```js
const obj = { property1: "abcdefg", property2: "hijklmno", property3: "pqrstuv", property4: "wxyz", property5: "Where going to need more alphabets!" };
```

**What does the rule currently do for this code?**
With allowAllPropertiesOnSameLine set to true there is no limit to line length.

**What will the rule do after it's changed?**
Additional parameter will only allow all properties on the same line up to a certain point.  Default 80 chars.

**Are you willing to submit a pull request to implement this change?**
I don't think I have the knowledge, no.

**Additional Info**
I can't seem to create similar rule using `max-len` in combination with existing rules.  We basically just want devs to break up object properties if the line length gets ridiculously long, but we don't want to limit line length for methods, functions, etc.

**Comment by anikethsaha** [comment]
Thanks for the report. 

`object-property-newline` is a stylistic rule and as per our new [policy](https://eslint.org/blog/2020/05/changes-to-rules-policies) all stylistic rules are frozen that means no new enhancement will be accepted atm. 

You can always create a custom rule that will extend this rule or implement a new rule.
